### PR TITLE
[RFC] Elements integration strategy

### PR DESCRIPTION
**Pull request by joshblack**
Hi all! 👋 Wanted to take a second to pull together an RFC for integration before proceeding with other packages. In this PR, I went ahead and pulled in the updated elements package with the corresponding namespace update. As a result, I ended up updating colors to match the old `ibm-color` signature and integrating grid.

During this integration, I started adding corresponding tests for the behavior that elements is replacing. This is so that we can guarantee that existing carbon behavior is not changed when the feature flag is on/off for `components-x`. Most notably this is for verifying that tokens exist and functions behave as expect / are defined.

I would love feedback from you all about this approach. At a high-level, the idea is to:

- Add `@carbon/elements` as a dependency
- Update a specific elements dependency
- Add tests for the classic behavior of the dependency, alongside the experimental implementation

For this PR specifically, I followed this pattern by doing the following:

- Add `@carbon/elements` as a dependency
- Update `_colors.scss` and `_theme.scss`/`_theme-tokens.scss`
  - This includes adding tests for colors and theme as a top-level export, verifying that tokens are defined with/without the feature flag set
- Update grid with new experimental implementation
  - This includes adding tests for classic grid implementation and asserting that this behavior does not change when experimental is included

In the future, we would expand this strategy to:

- Updating type
- Updating motion

Would love what you all think about this strategy, any of the test helpers, etc. to see what you think!

### CODE REVIEW

**Comment by joshblack** [comment]
Interesting files to look at for reviewiers:

- [Example test for grid](https://github.com/IBM/carbon-components/blob/598291e2645361118940415056426368ee5d9f7c/src/globals/grid/__tests__/grid-test.js)
- [Example test for colors](https://github.com/IBM/carbon-components/blob/598291e2645361118940415056426368ee5d9f7c/src/globals/scss/__tests__/colors-test.js)
- [Test helpers](https://github.com/IBM/carbon-components/blob/598291e2645361118940415056426368ee5d9f7c/tools/jest/scss.js)

**Review by asudoh — COMMENTED** [review]
Looks basically good to me - Thank you for doing this @joshblack! Could you please rebase to `master` as well as do the following updates (so Travis passes):

* Updating [`pure-modules-test.js`](https://github.com/IBM/carbon-components/blob/v9.70.0/tests/pure-modules-test.js#L22-L31) to ignore your test code from tree-shakability check
* Adding your test code to [the list of `.eslintignore`](https://github.com/IBM/carbon-components/blob/v9.70.0/.eslintignore#L24) for now (or updating ESLint rules)

**Comment by joshblack** [comment]
@asudoh I tried adding `jest: true` to `.eslintrc` but it seemed to have no effect (reference: https://eslint.org/docs/user-guide/configuring#specifying-environments). Any idea why?

**Review by mattrosno — COMMENTED** [review]
In the components, why are some of the updated variables prefixed with `carbon--` and others `ibm-color__`? E.g. `$carbon--gray-80` and `$ibm-color__gray-30`? Are these aliases of each other and if so, should we stick to the new namespace?

File: src/globals/scss/vendor/@carbon/elements/scss/layout/_convert.scss [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+// LICENSE file in the root directory of this source tree.
+//
+
+// Default, Use with carbon--em() and carbon--rem() functions
+$carbon--base-font-size: 16px !default;
+
+/// Convert a given px unit to a rem unit
+/// @param {px} $px
+/// @return {rem}
+@function carbon--rem($px) {
```
Comment (jnm2377) [inline src/globals/scss/vendor/@carbon/elements/scss/layout/_convert.scss:14]: It hadn't even occurred to me to add carbon to our function names. Will we be having to update our code where we're currently using `rem()`?

**Comment by joshblack** [comment]
@mattrosno I will update them all to the `carbon--` form in subsequent PRs 👍 Just was easier to map to `ibm-color` for now.

File: src/globals/scss/vendor/@carbon/elements/scss/layout/_convert.scss (hunk 1 above)
Comment (joshblack) [reply src/globals/scss/vendor/@carbon/elements/scss/layout/_convert.scss:14]: I think we can just keep using `rem` in the codebase and just alias out to the layout function, at least that's the hope!

**Review by jnm2377 — APPROVED** [review]
This looks great, Josh! Thanks for making these changes! 🎉

**Comment by joshblack** [comment]
First follow-up PR: https://github.com/IBM/carbon-components/pull/1804

**Comment by joshblack** [comment]
Second follow-up PR: https://github.com/IBM/carbon-components/pull/1809

**Comment by asudoh** [comment]
@joshblack Not certain, a couple guesses, though I haven't had time to verify each - One is such setting is a new feature of latest ESLint (this repo uses one major version older). Another is that it requires a plugin (as googling for that topic talks all about plugins).

**Comment by joshblack** [comment]
Final PR in #1823

**Pull request closed without merging**